services-camel/src/main/java/edu/unc/lib/dl/services/camel/BinaryEnhancementProcessor.java

```diff
@@ -52,23 +60,35 @@ public void process(final Exchange exchange) throws Exception {
             Element body = msgBody.getRootElement();
 
             String pidValue = body.getChild("pid", ATOM_NS).getTextTrim();
-            Element isEditThumbnail = body.getChild("editThumbnail", ATOM_NS);
 
-            if (isEditThumbnail != null) {
-                String mimeType = body.getChild("mimeType", ATOM_NS).getTextTrim();
+            String[] collThumbPath = pidValue.split("/");
+            String uuid = collThumbPath[collThumbPath.length - 1];
+            PID objPid = PIDs.get(uuid);
+
+            if (repoObjLoader.getCollectionObject(objPid) != null) {
```

Comment:
I think this will throw an error if the object pid isn't a collection. You would want to use `getRepositoryObject` and then check the type of the result. Also, if these methods don't find an object, they throw an exception rather than returning null.

However, I don't think i'd recommend the line `in.setHeader(FCREPO_RESOURCE_TYPE, Binary.getURI());` below since it's effectively declaring that a collection/folder/work/whatever is a binary for the sake of getting through the router.

My suggestion would be to populate `FCREPO_RESOURCE_TYPE` with the comma separated join of `repoObj.getTypes()`. Then add a new route here:
https://github.com/UNC-Libraries/Carolina-Digital-Repository/blob/65d5e3c120736859fb9c397d6b9ad4151213232d/services-camel/src/main/java/edu/unc/lib/dl/services/camel/enhancements/EnhancementRouter.java#L71

for processing non-binaries. In that route, there can be a new processor that does most of what you have in this `if` statement. This way, we don't need to give an inaccurate resource type, and it clearly separates the behavior for binaries vs others where they diverge. We also shouldn't need the `CdrEditThumbnail` header in this case. I think all the changes to BinaryEnhancementProcessor could be backed out except for the new stuff to set the resource types header. 

The non-binary route would be similar to the binary one. After it goes through the new processor (Maybe NonBinaryEnhancementProcessor), if enhancements are needed it should have CdrBinaryPath set, then you can set CdrEnhancementSet and call `"direct:process.enhancements", "direct-vm:solrIndexing"`. Otherwise, you'd just want to call solrIndexing.

A few other things:
1. the run enhancement message pid file should have a value you can just pass into `PIDs.get` without any extra splitting.
2. `thumbnailBasePath` should actually be the path for where the source image is, not where thumbnails would be. This should be updated in the ImportThumbnailService as well, and we should double check that it is using the correct directory.
3. Only set CdrBinaryPath for non-binaries if a file exists at the generated source image path. That way the router can decide whether to perform binary specific enhancements on if there is a binary.